libpod/container_internal_linux.go

```diff
@@ -788,16 +797,31 @@ func (c *Container) restore(ctx context.Context, options ContainerCheckpointOpti
 		if err := json.Unmarshal(networkJSON, &networkStatus); err != nil {
 			return err
 		}
-		// Take the first IP address
-		var IP net.IP
-		if len(networkStatus) > 0 {
-			if len(networkStatus[0].IPs) > 0 {
-				IP = networkStatus[0].IPs[0].Address.IP
+		if !options.IgnoreStaticIP {
+			// Take the first IP address
+			var IP net.IP
+			if len(networkStatus) > 0 {
+				if len(networkStatus[0].IPs) > 0 {
+					IP = networkStatus[0].IPs[0].Address.IP
+				}
+			}
+			if IP != nil {
+				// Tell CNI which IP address we want.
+				c.requestedIP = IP
 			}
 		}
-		if IP != nil {
-			// Tell CNI which IP address we want.
-			c.requestedIP = IP
+		if !options.IgnoreStaticMAC {
+			// Take the first MAC address
+			var MAC net.HardwareAddr
+			if len(networkStatus) > 0 {
+				if len(networkStatus[0].Interfaces) > 0 {
+					MAC, _ = net.ParseMAC(networkStatus[0].Interfaces[0].Mac)
```

Comment:
should we report the error here?

Reply:
I intentionally ignore the possible error because I think that the Mac string should be always ok here - it's not an input from user, we just need to change the type from String to HardwareAddr.

However, I do realize that it could be a pain to find out that why MAC is empty when there is a bug in `ioutil.ReadAll(networkStatusFile)` or `json.Unmarshal(networkJSON, &networkStatus)`.

Would the following code be good enough?

```go
MAC, err = net.ParseMAC(networkStatus[0].Interfaces[0].Mac)
if err != nill {
    return err
}
```

Reply:
As long as ParseMAC returns a decent Err message, otherwise do a 
errors.Wrapf(err, "Failed to ParseMAC %v", networkStatus[0].Interfaces[0].Mac)

Reply:
FYI

```
$ cat parsemac.go 
package main

import "fmt"
import "net"

func main() {
        var _, err = net.ParseMAC("fdaf")
        fmt.Printf("%s\n", err.Error())
}

$ go run parsemac.go 
address fdaf: invalid MAC address
```